app/controllers/sessions_controller.rb

```diff
@@ -22,12 +22,13 @@ def logout
 
   def home
   end
-  def profile 
-    if current_user.nil?
-       redirect_to :login 
+  
+  def profile
+    if current_user
+      redirect_to current_user
```

Comment:
Does current_user redirect there? I'm pretty sure the result of this method is ignored.